Model/Builders/ServiceOrderBuilder.php

```diff
@@ -205,66 +216,54 @@ protected function rstrpos($haystack, $needle, $offset = null)
         return $size - $pos - strlen($needle);
     }
 
-    /**
-     * @param $platformCode
-     * @param $issuer_id
-     *
-     * @return array
-     */
-
     public function getTransactions($platformCode, $issuer_id = null, $verifiedTermsOfService = null)
     {
-        return [
-            array_filter([
-                "payment_method"         => $platformCode,
-                "payment_method_details" => array_filter(
-                    [
-                        "issuer_id" => $issuer_id,
-                        "verified_terms_of_service" => $verifiedTermsOfService
-                    ]
-                )
-            ])
-        ];
+        return new Transaction(
+            paymentMethod: $platformCode,
+            paymentMethodDetails: new PaymentMethodDetails(
+                issuer_id: $issuer_id,
+                verified_terms_of_service: $verifiedTermsOfService
+            )
+        );
     }
 
     /**
      * @param $order
      * @param $paymentDetails
      * @param $customerData
-     *
+     * @param $urlProvider
      * @return Order
      */
-    public function collectData( $order, $paymentDetails, $customerData)
-    {
-        return new Order(
-            currency:  new Currency('EUR'),// new Currency($payment_amount->getCurrencyCode()),
-            amount: new Amount(floatval(1)*100),
-            transactions: $paymentDetails,
-            customer: $customerData
-        );
-    }
-
-    /**
-     * Collect data for order
-     *
-     * @return array
-     */
-
-    public function collectDataForOrder($order, $methodCode, $urlProvider, $orderLines, $paymentDetails, $customerData = null)
+    public function collectData($order, $paymentDetails, $customerData, $urlProvider, $paymentMethodDetails)
     {
-        $orderData = array_filter([
-            'amount' => $this->configRepository->getAmountInCents((float)$order->getBaseGrandTotal()),
-            'currency' => $order->getOrderCurrencyCode(),
-            'description' => $this->configRepository->getDescription($order, $methodCode),
-            'merchant_order_id' => $order->getIncrementId(),
-            'return_url' => $urlProvider->getReturnUrl(),
-            'webhook_url' => $urlProvider->getWebhookUrl(),
-            'transactions' => $paymentDetails,
-            'extra' => $this->getExtraLines(),
-            'order_lines' => $orderLines->get($order),
-            'customer' => $customerData
```

Comment:
Previously, the order had many different fields. It seems that after switching to the SDK, some fields disappeared.

```
description
merchant_order_id
extra
order_lines
...

```

Is it possible to flood them in order?

Reply:
Also `order_lines` is required for proper work of AfterPay and [KlarnaPayLater](https://emspay.stoplight.io/docs/online/68e1fdfbd1221-klarna-pay-later) . Could you check if it still work?